source/MaterialXShaderGen/Util.cpp

```diff
@@ -6,13 +6,51 @@
 #include <iostream>
 #include <sstream>
 #include <fstream>
-#if defined(_WIN32)
+
+#ifdef _WIN32
+#include <Windows.h>
 #include <fcntl.h>
+#else
+#include <dirent.h>
+#include <sys/types.h>
 #endif
 
 namespace MaterialX
 {
 
+void getDocumentsInDirectory(const std::string& directory, StringVec& files)
```

Comment:
Utility method for finding library documents in a directory.

Reply:
Good to have this at the util level. Seems like it's worth it to try and move this to core utils.